packages/zowe-explorer/src/commands/UnixCommandHandler.ts

```diff
@@ -43,219 +38,181 @@ export class UnixCommandHandler extends ZoweCommandProvider {
-            this.sshSession = new zosuss.SshSession(sshSessCfg);
-        } else {
-            ZoweLogger.info(this.unixCmdMsgs.opCancelledMsg);
-            Gui.showMessage(this.unixCmdMsgs.opCancelledMsg);
-        }
-    }
-
-    private async selectSshProfile(sshProfiles: imperative.IProfileLoaded[] = []): Promise<imperative.IProfileLoaded> {
-        ZoweLogger.trace("UnixCommandHandler.selectSshProfile called.");
-        let sshProfile: imperative.IProfileLoaded;
-        if (sshProfiles.length > 1) {
-            const sshProfileNamesList = sshProfiles.map((temprofile) => {
-                return temprofile.name;
-            });
-            if (sshProfileNamesList.length) {
-                const quickPickOptions: vscode.QuickPickOptions = {
-                    placeHolder: vscode.l10n.t("Select the ssh Profile."),
-                    ignoreFocusOut: true,
-                    canPickMany: false,
-                };
-                const sesName = await Gui.showQuickPick(sshProfileNamesList, quickPickOptions);
-                if (sesName === undefined) {
-                    Gui.showMessage(this.unixCmdMsgs.opCancelledMsg);
-                    return;
-                }
-
-                sshProfile = sshProfiles.filter((temprofile) => temprofile.name === sesName)[0];
+                await AuthUtils.errorHandling(error, this.nodeProfile.name);
             }
-        } else if (sshProfiles.length > 0) {
-            sshProfile = sshProfiles[0];
         }
-        return sshProfile;
     }
 
     private async getSshProfile(): Promise<void> {
-        ZoweLogger.trace("UnixCommandHandler.getsshProfile called.");
         const profiles = await this.profileInstance.fetchAllProfilesByType("ssh");
         if (!profiles.length) {
             ZoweLogger.error(this.unixCmdMsgs.sshProfNotFoundMsg);
             Gui.errorMessage(this.unixCmdMsgs.sshProfNotFoundMsg);
             return;
         }
-        if (profiles.length > 0) {
-            this.sshProfile = await this.selectSshProfile(profiles);
-            if (!this.sshProfile) {
-                return;
-            }
-            if (!(this.sshProfile.profile.host && this.sshProfile.profile.port)) {
-                const currentProfile = await this.profileInstance.getProfileFromConfig(this.sshProfile.name);
-                const filePath = currentProfile.profLoc.osLoc[0];
-                await this.profileInstance.openConfigFile(filePath);
-                ZoweLogger.error(this.unixCmdMsgs.sshProfMissingInfoMsg);
-                Gui.errorMessage(this.unixCmdMsgs.sshProfMissingInfoMsg);
+        this.sshProfile = await this.selectServiceProfile(profiles);
+        if (!this.sshProfile) {
+            return;
+        }
+        if (!(this.sshProfile.profile.host && this.sshProfile.profile.port)) {
+            const currentProfile = await this.profileInstance.getProfileFromConfig(this.sshProfile.name);
+            const filePath = currentProfile.profLoc.osLoc[0];
```

Comment:
I know its highly unlikely that `osLoc` will be undefined, but should we use optional chaining on the array access so its not flagged by Coverity?

```suggestion
            const filePath = currentProfile.profLoc.osLoc?.[0];
```